tests/benchmark/bm_spaces.cpp

```diff
@@ -20,22 +20,49 @@
         }                                                                                          \
     }
 
+// Defining the generic benchmark flow: if there is support for the optimization, benchmark the
+// function.
+#define BENCHMARK_DISTANCE_FP64_F(arch, settings, func)                                            \
+    BENCHMARK_DEFINE_F(BM_VecSimSpaces_FP64, arch##_##settings)(benchmark::State & st) {           \
+        if (opt < ARCH_OPT_##arch) {                                                               \
+            st.SkipWithError("This benchmark requires " #arch ", which is not available");         \
+            return;                                                                                \
+        }                                                                                          \
+        for (auto _ : st) {                                                                        \
+            func(v1, v2, dim);                                                                     \
+        }                                                                                          \
+    }
+
 // AVX512 functions
 #ifdef __AVX512F__
 #include "VecSim/spaces/L2/L2_AVX512.h"
 #include "VecSim/spaces/IP/IP_AVX512.h"
 
-BENCHMARK_DISTANCE_F(AVX512, L2_16, FP32_L2SqrSIMD16Ext_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, L2_4, FP32_L2SqrSIMD4Ext_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, L2_16_Residuals, FP32_L2SqrSIMD16ExtResiduals_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, L2_4_Residuals, FP32_L2SqrSIMD4ExtResiduals_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, L2_16, FP32_L2SqrSIMD16Ext_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, L2_4, FP32_L2SqrSIMD4Ext_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, L2_16_Residuals, FP32_L2SqrSIMD16ExtResiduals_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, L2_4_Residuals, FP32_L2SqrSIMD4ExtResiduals_AVX512)
+
+BENCHMARK_DISTANCE_F(AVX512_F, IP_16, FP32_InnerProductSIMD16Ext_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, IP_4, FP32_InnerProductSIMD4Ext_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, IP_16_Residuals, FP32_InnerProductSIMD16ExtResiduals_AVX512)
+BENCHMARK_DISTANCE_F(AVX512_F, IP_4_Residuals, FP32_InnerProductSIMD4ExtResiduals_AVX512)
+
+BENCHMARK_DISTANCE_FP64_F(AVX512_F, IP_2, FP64_InnerProductSIMD2Ext_AVX512_noDQ)
+BENCHMARK_DISTANCE_FP64_F(AVX512_F, IP_2_Residuals, FP64_InnerProductSIMD2ExtResiduals_AVX512_noDQ)
 
-BENCHMARK_DISTANCE_F(AVX512, IP_16, FP32_InnerProductSIMD16Ext_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, IP_4, FP32_InnerProductSIMD4Ext_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, IP_16_Residuals, FP32_InnerProductSIMD16ExtResiduals_AVX512)
-BENCHMARK_DISTANCE_F(AVX512, IP_4_Residuals, FP32_InnerProductSIMD4ExtResiduals_AVX512)
+BENCHMARK_DISTANCE_FP64_F(AVX512_F, L2_2, FP64_L2SqrSIMD2Ext_AVX512_noDQ)
+BENCHMARK_DISTANCE_FP64_F(AVX512_F, L2_2_Residuals, FP64_L2SqrSIMD2ExtResiduals_AVX512_noDQ)
```

Comment:
we need a todo to add the 8 varient, right?

Reply:
Added comment